src/rollup/index.ts

```diff
@@ -211,45 +212,81 @@ export default function rollup(rawInputOptions: GenericConfigObject): Promise<Ro
 								optimized = true;
 							}
 
-							assignChunkIds(chunks, inputOptions, outputOptions, inputBase, addons);
-
-							// assign to outputBundle
-							for (let i = 0; i < chunks.length; i++) {
-								const chunk = chunks[i];
-								const facadeModule = chunk.facadeModule;
-
-								outputBundle[chunk.id] = {
-									code: undefined as any,
-									dynamicImports: chunk.getDynamicImportIds(),
-									exports: chunk.getExportNames(),
-									facadeModuleId: facadeModule && facadeModule.id,
-									fileName: chunk.id,
-									imports: chunk.getImportIds(),
-									isDynamicEntry:
-										facadeModule !== null && facadeModule.dynamicallyImportedBy.length > 0,
-									isEntry: facadeModule !== null && facadeModule.isEntryPoint,
-									map: undefined,
-									modules: chunk.renderedModules,
-									get name() {
-										return chunk.getChunkName();
-									}
-								} as OutputChunk;
-							}
-
 							return Promise.all(
 								chunks.map(chunk => {
-									const outputChunk = outputBundle[chunk.id] as OutputChunk;
-									return chunk.render(outputOptions, addons, outputChunk).then(rendered => {
-										outputChunk.code = rendered.code;
-										outputChunk.map = rendered.map;
-
-										return graph.pluginDriver.hookParallel('ongenerate', [
-											{ bundle: outputChunk, ...outputOptions },
-											outputChunk
-										]);
-									});
+									const facadeModule = chunk.facadeModule;
+									const chunkInfo = {
```

Comment:
I do not like calculating this information twice. Besides performance considerations, this information will need to be maintained and synced in more than one place which is always problematic.

One way forward might be to first create a Map of Chunk -> PreRenderedChunk info and use this map in the second usage.

Reply:
Sure, could also do it in an array and just match positions.

Reply:
I think that would be fine as well.